app/master.js

```diff
@@ -48,7 +49,18 @@ class Master {
       Master.activateFileListener(watcher);
     }
 
-    // Fork workers
+    // Clear background job locks
+    return Master._initAddons()
+      .then(() => Master._forkWorkers);
+  }
+
+  static _initAddons() {
+    return AddonManager.readAddons()
+      .then(() => AddonManager.clearJobLocks())
+      .then(() => AddonManager.installAddons());
```

Comment:
this cannot be here in master - otherwise addons dependencies cannot update without restarting master which shouldn't be required in this case